source/FilesServer/app/service.py

```diff
@@ -22,10 +22,25 @@ def is_file_exists(
     return True
 
 
-async def s3_get_presigned_url(
-    file_name: str | None, s3_client: boto3.client = Depends(get_s3_client)
-) -> str | None:
+def s3_get_head_object_http_headers(file_name: str) -> dict | None:
     settings = get_settings()
+    s3_client = get_s3_client()
+    if not is_file_exists(file_name):
+        return None
+    return s3_client.head_object(Bucket=settings.aws_bucket_name, Key=file_name)[
+        "ResponseMetadata"
+    ]["HTTPHeaders"]
+
+
+async def s3_get_metadata(file_name: str) -> FileMetadata | None:
+    http_headers = s3_get_head_object_http_headers(file_name)
```

Comment:
same, you need to handle the optional None. maybe raising an exception will be easier. otherwise you need to check for None after every call

Reply:
handling in endpoint